index.js

```diff
@@ -75,6 +88,36 @@ Filter.prototype.build = function build() {
   });
 };
 
+/*
+ * @private
+ *
+ *
+ * @method cachKey
+ * @return {String} this filters top-level cache key
+ */
+Filter.prototype.cacheKey = function() {
+  return hashForDep(this.baseDir());
```

Comment:
As this is fairly costly, and we easily have N independent broccoli transforms. I believe this only needs to run once per subclass, not once per instance. As such I believe caching the result on the constructor should be sufficient.